aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -121,14 +122,13 @@ Completable hydrate() {
      * @param lastSyncTime The time of a last successful sync.
      * @return The input, or {@link SyncTime#never()}, if the last sync time is "too old."
      */
-    private SyncTime filterOutOldSyncTimes(SyncTime lastSyncTime) throws DataStoreException {
-        if (!lastSyncTime.exists()) {
+    private SyncTime filterOutOldSyncTimes(SyncTime lastSyncTime) {
+        if (!lastSyncTime.exists() || dataStoreConfiguration == null) {
```

Comment:
I'd expect that `dataStoreConfiguration` would be guaranteed non-null, here. That is, if there's no config file, and the user supplies no config, then you still have some system defaults you can use. Right? So, _those_ could go into a non-null `dataStoreConfiguration`. And that would have a different effect here.

The current logic says basically "never do a delta sync."

But that's not what we want -- we want a delta sync if the last sync time is within some san default (1 day? Whatever it is?)

In other words, I'd prefer to have an `Objects.requireNonNull(...)` and `@NonNull` directive on the `SyncProcessor` constructor, and then just never have to worry about this branch, in this spot.

Reply:
Yeah. I added this here to prevent a potential NPE, but by the time it gets here, we’re sure to have at least default values.

Reply:
Changes made in the [latest commit](https://github.com/aws-amplify/amplify-android/pull/401/commits/6c3c4c5522dcd21c055d511c03afb970e64bede5)

I created a provider interface that I'm now passing from the the plugin down to the `Orchestrator` and `SyncProcessors`. By the time these classes invoke the provider method to get the configuration, the plugin will have already setup the configuration in the `configure` method.